pkg/kubelet/cm/cpumanager/state/state_file.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"encoding/json"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"os"
+	"sync"
+)
+
+type stateFileData struct {
+	DefaultCPUSet string            `json:"defaultCpuSet"`
+	Entries       map[string]string `json:"reservedList,omitempty"`
+}
+
+type stateFile struct {
+	sync.RWMutex
+	stateFilePath string
+	State
+}
+
+// NewFileState creates new State for keeping track of cpu/pod assignment with file backend
+func NewFileState(filePath string) State {
+	stateFile := &stateFile{
+		stateFilePath: filePath,
+		State:         NewMemoryState(),
+	}
+	stateFile.tryRestoreState()
+	return stateFile
+}
+
+// tryRestoreState tries to read state file, upon any error,
+// err message is logged and state is left clean. un-initialized
+func (sf *stateFile) tryRestoreState() {
```

Comment:
Could we return an `error` from this function? Then I think you could clean up a lot of the redundant condition checks by bailing as soon as you fail to init the file. At the call site, if err is not nil then you can just call `storeState()`. We may want another check in `NewFileState` after all that to simply `stat` the file and panic if it still doesn't exist.

Reply:
:+1: